core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeService.java

```diff
@@ -92,9 +98,8 @@ public DiscoveryNode buildLocalNode(TransportAddress publishAddress) {
                 logger.warn("failed to build custom attributes from provider [{}]", e, provider);
```

Comment:
`CustomAttributesProvider` gave me a hard-attack, can you open an issue to move the registration to a module please?